src/Common/Abstracts/Custom_Table_Repository.php

```diff
@@ -0,0 +1,1081 @@
+<?php
+/**
+ * The custom table repository.
+ *
+ * @since TBD
+ *
+ * @package TEC\Common\Abstracts
+ */
+
+declare( strict_types=1 );
+
+namespace TEC\Common\Abstracts;
+
+use TEC\Common\Contracts\Custom_Table_Repository_Interface as Repository_Interface;
+use TEC\Common\StellarWP\SchemaModels\Contracts\SchemaModel as Model;
+use TEC\Common\StellarWP\Schema\Tables\Contracts\Table as Table_Interface;
+use TEC\Common\StellarWP\SchemaModels\Relationships\ManyToManyWithPosts;
+use TEC\Common\StellarWP\DB\DB;
+use Tribe__Promise as Promise;
+use RuntimeException;
+use Generator;
+
+/**
+ * The custom table repository.
+ *
+ * @since TBD
+ *
+ * @package TEC\Common\Abstracts
+ */
+abstract class Custom_Table_Repository implements Repository_Interface {
+	/**
+	 * The schema.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	private array $schema = [];
+
+	/**
+	 * The update fields aliases.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	private array $aliases = [];
+
+	/**
+	 * The default arguments.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	private array $default_select_args = [];
+
+	/**
+	 * The default arguments used for create queries.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	private array $default_create_args = [];
+
+	/**
+	 * The arguments used for select queries.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	private array $select_args = [];
+
+	/**
+	 * The arguments used for upsert queries.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	private array $upsert_args = [];
+
+	/**
+	 * The found rows.
+	 *
+	 * @since TBD
+	 *
+	 * @var int
+	 */
+	private int $found_rows = 0;
+
+	/**
+	 * The page used for select queries.
+	 *
+	 * @since TBD
+	 *
+	 * @var int
+	 */
+	private int $page = 1;
+
+	/**
+	 * The per page used for select queries.
+	 *
+	 * If not changed, we are selecting them all by default.
+	 *
+	 * @since TBD
+	 *
+	 * @var int
+	 */
+	private int $per_page = 10000000000;
+
+	/**
+	 * The fields used for select queries.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	private array $fields = [ '*' ];
+
+	/**
+	 * The schema callbacks.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	private array $schema_callbacks = [];
+
+	/**
+	 * Whether to use the default arguments.
+	 *
+	 * @since TBD
+	 *
+	 * @var bool
+	 */
+	private bool $use_default_args = true;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since TBD
+	 */
+	public function __construct() {
+		$this->order_by( $this->get_table_interface()::uid_column(), 'DESC' );
+
+		$operators = $this->get_table_interface()::operators();
+
+		foreach ( $this->get_table_interface()::get_columns()->get_names() as $column ) {
+			foreach ( $operators as $operator_slug => $operator ) {
+				if ( 'eq' === $operator_slug ) {
+					$this->add_schema_entry(
+						$column,
+						function ( $value ) use ( $column, $operator ) {
+							return [
+								'column'   => $column,
+								'value'    => $value,
+								'operator' => $operator,
+							];
+						}
+					);
```

Comment:
Should there be a `continue` statement here? Or do we intend to add a second schema entry?